pkg/virt-launcher/virtwrap/cli/event.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package cli
+
+import (
+	"fmt"
+
+	"libvirt.org/libvirt-go"
+)
+
+type DomainEventDeviceRemovedRegistrar struct {
+	connect        *libvirt.Connect
+	domain         *libvirt.Domain
+	registrationID int
+}
+
+func NewDomainEventDeviceRemovedRegistrar(connection Connection, virDomain VirDomain) *DomainEventDeviceRemovedRegistrar {
+	libvirtConnect := extractLibvirtConnect(connection)
+	libvirtDomain := extractLibvirtDomain(virDomain)
+	if libvirtConnect != nil && libvirtDomain != nil {
+		return &DomainEventDeviceRemovedRegistrar{
+			connect: libvirtConnect,
+			domain:  libvirtDomain,
+		}
+	}
+	return nil
+}
+
+func (r *DomainEventDeviceRemovedRegistrar) Register(callback interface{}) error {
+	f, ok := callback.(libvirt.DomainEventDeviceRemovedCallback)
```

Comment:
It can be obvious to one but not trivial to others.
our code base already full with shorten names that only one who is familiar with the lingo could understand
Also that habit of keep the first argument of a function to be one character dont make things easier.

Reply:
The brain power to map `f, ok := callback.(libvirt.DomainEventDeviceRemovedCallback)` to `id, err := r.connect.DomainEventDeviceRemovedRegister(r.domain, f)` a few lines below is minimal.

It is common in Go to use short names when appropiate. This is true when naming the receiver and when naming short scoped variables that their purpose is clear from the context (in our case, the extracted type).